include/cooperative_perception/angle.hpp

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2022 Leidos
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef COOPERATIVE_PERCEPTION_ANGLE_HPP
+#define COOPERATIVE_PERCEPTION_ANGLE_HPP
+
+#include <complex>
+#include <math.h>
+#include <units.h>
+#include <cooperative_perception/utils.hpp>
+
+namespace cooperative_perception
+{
+/**
+ * @brief Angle type for representing and manipulating angles
+ */
+class Angle
+{
+private:
+  std::complex<double> value_{};
+
+public:
+  Angle(units::angle::radian_t angle_val) noexcept : value_{ units::math::cos(angle_val), units::math::sin(angle_val) }
+  {
+  }
+
+  auto set_angle(units::angle::radian_t angle_val) noexcept -> void
+  {
+    value_ = std::complex<double>{ units::math::cos(angle_val), units::math::sin(angle_val) };
```

Comment:
This should use the `real(...)` and `imag(...)` setters to modify the existing `value_` object instead of reassigning it to a newly-created one. It should look like:
```cpp
value_.real(units::math::cos(angle_val));
value_.imag(units::math::sin(angle_val));
```

Reply:
Makes sense - creating a new one of everyone `set_angle` is too much!